uk_geo_utils/base_updater.py

```diff
@@ -0,0 +1,373 @@
+import abc
+import shutil
+import tempfile
+import urllib.request
+import zipfile
+from pathlib import Path
+
+import psutil
+from django.core.management.base import BaseCommand
+from django.db import connection, transaction
+
+
+def unzip(filepath):
+    zip_file = zipfile.ZipFile(filepath, "r")
+    tmpdir = tempfile.mkdtemp()
+    zip_file.extractall(tmpdir)
+    return tmpdir
+
+
+def check_memory(required_memory: int = 2):
+    # Downloading, unzipping and working with the ONSPD
+    # requires a decent chunk of memory to play with.
+    # Running this import on a tiny instance like a
+    # t2.micro will cause an Out Of Memory error
+
+    # By default ensure we've got >2Gb total before we start
+    mem = psutil.virtual_memory()
+    gb = ((mem.total / 1024) / 1024) / 1024
+    return gb >= required_memory
+
+
+class BaseUpdater(BaseCommand):
```

Comment:
Bit of a nitpick, but I reckon we can call the base class `BaseImporter` now there's no distinction between "import" and "update"